cdap-examples/DataCleansing/src/main/java/co/cask/cdap/examples/datacleansing/DataCleansingMapReduce.java

```diff
@@ -0,0 +1,152 @@
+    context.setInput(DataCleansing.RAW_RECORDS, rawRecords);
+
+    // Each run writes its output to a partition for the league
+    Long timeKey = Long.valueOf(context.getRuntimeArguments().get(OUTPUT_PARTITION_KEY));
+    PartitionKey outputKey = PartitionKey.builder().addLongField("time", timeKey).build();
+    Map<String, String> outputArgs = new HashMap<>();
+    PartitionedFileSetArguments.setOutputPartitionKey(outputArgs, outputKey);
+
+    PartitionedFileSet outputFileSet = context.getDataset(DataCleansing.CLEAN_RECORDS, outputArgs);
+    context.setOutput(DataCleansing.CLEAN_RECORDS, outputFileSet);
+
+    Job job = context.getHadoopJob();
+    job.setMapperClass(SchemaMatchingFilter.class);
+    job.setReducerClass(IdentityReducer.class);
+
+    // simply propagate the schema (if any) to be used by the mapper
+    String schemaJson = context.getRuntimeArguments().get(SCHEMA_KEY);
+    if (schemaJson != null) {
+      job.getConfiguration().set(SCHEMA_KEY, schemaJson);
+    }
+  }
+
+  @Override
+  public void onFinish(boolean succeeded, MapReduceContext context) throws Exception {
+    if (succeeded) {
+      partitionConsumer.persist(context);
+    }
+  }
+
+  /**
+   * A Mapper which skips text that doesn't match a given schema.
+   */
+  public static class SchemaMatchingFilter extends Mapper<LongWritable, Text, LongWritable, Text> {
+    public static final Schema DEFAULT_SCHEMA = Schema.recordOf("person",
+                                                                Schema.Field.of("pid", Schema.of(Schema.Type.LONG)),
+                                                                Schema.Field.of("name", Schema.of(Schema.Type.STRING)),
+                                                                Schema.Field.of("dob", Schema.of(Schema.Type.STRING)),
+                                                                Schema.Field.of("zip", Schema.of(Schema.Type.INT)));
+
+    private SimpleSchemaMatcher schemaMatcher;
+    private Metrics mapMetrics;
+
+    @Override
+    protected void setup(Context context) throws IOException, InterruptedException {
+      // setup the schema to be used by the mapper
+      String schemaJson = context.getConfiguration().get(SCHEMA_KEY);
+      if (schemaJson == null) {
+        schemaMatcher = new SimpleSchemaMatcher(DEFAULT_SCHEMA);
+      } else {
+        schemaMatcher = new SimpleSchemaMatcher(Schema.parseJson(schemaJson));
+      }
+    }
+
+    @Override
+    public void map(LongWritable key, Text data, Context context) throws IOException, InterruptedException {
+      if (!schemaMatcher.matches(data.toString())) {
+        mapMetrics.count("data.invalid", 1);
+        return;
+      }
+      context.write(key, data);
+    }
+  }
+
+  /**
+   * An identity reducer which writes only the values to the file.
+   */
+  public static class IdentityReducer extends Reducer<LongWritable, Text, NullWritable, Text> {
```

Comment:
with a map-only task, the records are not the only thing written to the output file.
This IdentityReducer ensures that only the values are written to the file (so as to match the format of the input file).